samtranslator/intrinsics/resolver.py

```diff
@@ -17,7 +18,9 @@ def __init__(self, parameters, supported_intrinsics=DEFAULT_SUPPORTED_INTRINSICS
         """
 
         if parameters is None or not isinstance(parameters, dict):
-            raise TypeError("parameters must be a valid dictionary")
```

Comment:
Is `TypeError` causes 500 errors?

Reply:
sorry, what?

Reply:
If we raise `TypeError` instead of `InvalidDocumentException`, will it cause 500 server error on lambda execution? Is that the reason we changed the type of the exception here?

Reply:
ohh i see. yes TypeError causes 500 server error.